ddl/ddl_api.go

```diff
@@ -6900,7 +6900,6 @@ func BuildHiddenColumnInfo(ctx sessionctx.Context, indexPartSpecifications []*as
 				return nil, errors.Trace(err)
 			}
 		}
-		idxPart.Expr = nil
```

Comment:
We should remove this line to keep `IndexPartSpecification` immutable because here: https://github.com/pingcap/tidb/blob/96dcbb7c9a20f417fe514d0c0a21abfae553f0d7/ddl/schematracker/checker.go#L220-L232 will call `CreateTable` twice.

Reply:
I wonder if there's any bug when this is not removed?

Reply:
I'm not sure whether we should call it bug, beacause it only happens in test cases.